### ISSUE TITLE
can not exec func “handle” of subscriber

### ISSUE THREAD

**Issue by HelloZouYou** [micro/go-plugins#423]
Thanks for reading.

service type: srv
broker: nsq

Service runs successfully.It shows like:

```
2019-11-19 16:17:57.955413 I | Transport [http] Listening on [::]:43547
INF    1 (nsqd:4150) connecting to nsqd
2019-11-19 16:17:57.970075 I | Broker [nsq] Connected to nsqd:4150
2019-11-19 16:17:57.970716 I | Registry [etcd] Registering node: go.micro.srv.http-ee90a9f4-7997-449a-baaf-9720206ff818
INF    2 [send.httpmsg/85a242e1-9caf-4b53-8588-375cba27f5e7#ephemeral] (nsqd:4150) connecting to nsqd
2019-11-19 16:17:57.987230 I | Subscribing go.micro.srv.http-ee90a9f4-7997-449a-baaf-9720206ff818 to topic: httpmsg
```

But there was an error when I pushed a message to the queue.
Error log:

```
panic: assignment to entry in nil map
goroutine 70 [running]:
github.com/micro/go-micro/server.(*rpcServer).createSubHandler.func1(0x1661680, 0xc00042fc50, 0x0, 0x0)
        /go/pkg/mod/github.com/micro/go-micro@v1.15.1/server/subscriber.go:177 +0xb41
github.com/micro/go-plugins/broker/nsq.(*nsqBroker).Subscribe.func1(0xc0004542a0, 0x1, 0x101)
        /go/pkg/mod/github.com/micro/go-plugins@v1.4.0/broker/nsq/nsq.go:268 +0x122
github.com/nsqio/go-nsq.HandlerFunc.HandleMessage(0xc00042f680, 0xc0004542a0, 0x1388860, 0xc00042f680)
        /go/pkg/mod/github.com/nsqio/go-nsq@v1.0.7/consumer.go:42 +0x30
github.com/nsqio/go-nsq.(*Consumer).handlerLoop(0xc000243200, 0x1644400, 0xc00042f680)
        /go/pkg/mod/github.com/nsqio/go-nsq@v1.0.7/consumer.go:1119 +0x13d
created by github.com/nsqio/go-nsq.(*Consumer).AddConcurrentHandlers
        /go/pkg/mod/github.com/nsqio/go-nsq@v1.0.7/consumer.go:1101 +0x7b
exit status 2
```

It is certain that the message has been successfully pushed to NSQ，but it cannot be processed correctly.

**Comment by HelloZouYou** [comment]
I got it. I use `POSTMAN` to push messages, so there is no HEADER and BODY in the structure.And there is no assignment after JSON Unmarshal.

**Comment by vtolstov** [comment]
i think that micro must protect this(i'm fix the code), but you must send proper message.

**Comment by vtolstov** [comment]
fix in #963
